dpkgfiles/make_deb_package.py

```diff
@@ -76,12 +108,88 @@ def pwd():
 execAndWait('rm -rf %s' % pkgdir)
 execAndWait('rm -f %s*' % pkgdir)
 execAndWait('rm -f %s*' % pkgdir_)
-shutil.copytree(origDir, pkgdir, ignore=shutil.ignore_patterns('__pycache__'))
-execAndWait('tar -zcf %s.tar.gz %s' % (pkgdirOrig, pkgdir))
+shutil.copytree(origDir, pkgdir, ignore=shutil.ignore_patterns('depends'))
+
+faketimePath = find('libfaketime.so.1', '/usr/lib')
+faketimeVars = 'export LD_PRELOAD=%s; export FAKETIME="2013-06-01 00:00:00";' % faketimePath
+
+execAndWait('%s tar -zcf %s.tar.gz %s' % (faketimeVars, pkgdir, pkgdir))
 cd(pkgdir)
-execAndWait('export DEBFULLNAME="Armory Technologies, Inc."; dh_make -s -e support@bitcoinarmory.com -f ../%s.tar.gz' % pkgdirOrig)
+# piping yes into dh_make causes us to automatically move past the prompt
+# to verify that the data is what we expect it to be
+execAndWait('%s export DEBFULLNAME="Armory Technologies, Inc."; yes | dh_make -s -e support@bitcoinarmory.com -f ../%s.tar.gz' % (faketimeVars, pkgdir))
 for f in dpkgfiles:
-   shutil.copy('dpkgfiles/%s' % f, 'debian/%s' % f)
+   execAndWait('%s cp dpkgfiles/%s debian/%s' % (faketimeVars, f, f))
 
 # Finally, all the magic happens here
-execAndWait('dpkg-buildpackage -rfakeroot')
+execAndWait('%s pdebuild --pbuilder cowbuilder --use-pdebuild-internal --architecture %s --buildresult ../armory-build -- --basepath /var/cache/pbuilder/%s' % (faketimeVars, arch, args.chroot))
+
+# Download deb packages of dependencies for offline bundle
+if args.depends:
+    cd('dpkgfiles')
+    execAndWait('mkdir packages')
+    cd('packages')
+    # apt/dpkg require a lot of specific directories/files to function
+    execAndWait('mkdir parts')
+    execAndWait('mkdir -p var/lib/apt/lists/partial')
+    execAndWait('mkdir -p var/cache/apt/archives/partial')
+    execAndWait('mkdir -p var/lib/dpkg')
+    execAndWait('touch var/lib/dpkg/status')
+    # remove any debs left over from last time, because this is just our
+    # temporary working directory
+    execAndWait('rm -f *.deb')
+
+    depends = ['libqt4-designer', 'libqt4-help', 'libqt4-scripttools',
+            'libqt4-test', 'libqtassistantclient4', 'libqtwebkit4',
+            'python-psutil', 'python-pyasn1', 'python-qt4', 'python-sip',
+            'python-twisted', 'python-twisted-conch', 'python-twisted-lore',
+            'python-twisted-mail', 'python-twisted-news',
+            'python-twisted-runner', 'python-twisted-words']
+    execAndWait('apt-get -c etc/apt.conf update')
+    for deb in depends:
+        execAndWait('apt-get -c etc/apt.conf download %s:%s' % (deb, arch))
+    
+    print 'All dependency debs were downloaded'
```

Comment:
I'm pretty sure that ap-get can fail when doing the update or the downloads. There doesn't appear to be a way to get the error code from execAndWait, so I think I need to modify it slightly to return the code.